atlasdb-client/src/main/java/com/palantir/atlasdb/util/AtlasDbMetrics.java

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright 2017 Palantir Technologies
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.atlasdb.util;
+
+import java.util.concurrent.TimeUnit;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.codahale.metrics.JmxReporter;
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.SharedMetricRegistries;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+
+public final class AtlasDbMetrics {
+    private static final Logger log = LoggerFactory.getLogger(AtlasDbMetrics.class);
+    private static final String DEFAULT_DOMAIN = "com.palantir.util";
+    @VisibleForTesting
+    static final String DEFAULT_REGISTRY_NAME = "AtlasDb";
+
+    @VisibleForTesting
+    static volatile MetricRegistry metrics;
+
+    private AtlasDbMetrics() {}
+
+    public static synchronized void setMetricRegistry(MetricRegistry metricRegistry) {
+        AtlasDbMetrics.metrics = Preconditions.checkNotNull(metricRegistry, "Metric registry cannot be null");
+    }
+
+    // Using this means that all atlasdb clients will report to the same registry, which may give confusing stats
+    public static MetricRegistry getMetricRegistry() {
+        if (AtlasDbMetrics.metrics == null) {
+            synchronized (AtlasDbMetrics.class) {
+                if (AtlasDbMetrics.metrics == null) {
+                    AtlasDbMetrics.metrics = SharedMetricRegistries.getOrCreate(DEFAULT_REGISTRY_NAME);
+                    JmxReporter.forRegistry(AtlasDbMetrics.metrics)
```

Comment:
I don't think we should create the MetricRegistry in AtlasDB, I think we should require it be injected from outside (e.g. via Dropwizard), where the consumers have specified how they want metrics reported (e.g. via JMX, logging, HTTP endpoint, etc.)